### DISCUSSION TITLE
How to get template parameters

### DISCUSSION THREAD

**Discussion by DongsGit** [artifacthub/hub#1759] [Q&A] [answered]
I want to extract the parameters from the template,Such as String, List, Map and other data structures.Arrange the parameters like this
![image](https://user-images.githubusercontent.com/36791760/146711405-0e4897f8-577d-4a46-94ee-4ece0a8b4beb.png)

**Comment by tegioz** [comment, marked as answer]
Hi @DongsGit 👋 

Each chart contains a `values.yaml` file with the default configuration values. For this particular case, you can find the default values file [here](https://github.com/mogaal/helm-charts/blob/main/prometheus-yace-exporter/values.yaml). One option could be to parse that `yaml` file. You could also use a regular expression to extract them from the templates files though.

**Reply by DongsGit** [reply]
First of all, thank you for your help.I know regular expressions, and I have implemented some requirements through them.Thank you very munch.

**Reply by tegioz** [reply]
No worries 👍
